File: queu-front/src/types/ticket.ts
export interface Ticket {
  id: number
  ticketCode: string
  queueId: number
  citizenId: number
  ticketStatus: TicketStatus
  priority: TicketPriority
  issuedAt: string
  calledAt?: string
  startedAt?: string
  completedAt?: string
  cancelledAt?: string
  createdAt: string
  updatedAt?: string
  // Relacionamentos opcionais para mostrar dados relacionados
  queue?: {
    id: number
    name: string
    code: string
  }
  citizen?: {
    id: number
    name: string
    document: string
  }
}

export enum TicketStatus {
  Waiting = 0, // Aguardando chamada
  Called = 1, // Chamado
  InService = 2, // Em atendimento
  Completed = 3, // Atendimento concluído
  Cancelled = 4, // Cancelado
  NoShow = 5, // Não compareceu
}

export enum TicketPriority {
  Normal = 0, // Prioridade normal
  Priority = 1, // Prioridade alta
  Urgent = 2, // Urgente
  Late = 3, // Atrasado
}

export interface CreateTicketDto {
  queueId: number
  citizenId: number
  priority: TicketPriority
}

export interface UpdateTicketDto {
  id: number
  ticketNumber: string
  ticketStatus: TicketStatus
  priority: TicketPriority
}

export interface NextTicketDto {
  queueId: number
}

export interface ApiResponse<T = any> {
  success: boolean
  message: string
  data: T
  errors?: string[]
  statusCode: number
  timestamp: string
}

export interface TicketListResponse {
  tickets: Ticket[]
  total: number
  page?: number
  limit?: number
}

File: queue-back/src/Domain/Entities/Ticket.cs
using cronly_back.Domain.Enums;

namespace cronly_back.Domain.Entities;

public class Ticket
{
    public int Id { get; set; }
    public string TicketCode { get; set; } = string.Empty;
    public int QueueId { get; set; }
    public int CitizenId { get; set; }
    public TicketStatus TicketStatus { get; set; } = TicketStatus.Waiting;
    public TicketPriority Priority { get; set; } = TicketPriority.Normal;
    
    public DateTime IssuedAt { get; set; } = DateTime.UtcNow;
    public DateTime? CalledAt { get; set; }
    public DateTime? StartedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public DateTime? CancelledAt { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    public Queue Queue { get; set; } = null!;
    public Citizen Citizen { get; set; } = null!;
    
    public Ticket(){}
    
    public Ticket(string ticketCode, int queueId, int citizenId, TicketStatus ticketStatus, TicketPriority priority)
    {
        TicketCode = ticketCode;
        QueueId = queueId;
        CitizenId = citizenId;
        TicketStatus = ticketStatus;
        Priority = priority;
        CreatedAt = DateTime.UtcNow;   
        IssuedAt = DateTime.UtcNow;   
    }
    
    public void Update(string ticketNumber, TicketStatus ticketStatus, TicketPriority priority)
    {
        TicketCode = ticketNumber;
        TicketStatus = ticketStatus;
        Priority = priority;
        UpdatedAt = DateTime.UtcNow;   
    }
}